pkg/containertools/runner.go

```diff
@@ -22,14 +22,75 @@ type CommandRunner interface {
 type ContainerCommandRunner struct {
 	logger        *logrus.Entry
 	containerTool ContainerTool
+	config        *RunnerConfig
+}
+
+type RunnerConfig struct {
+	SkipTLS bool
+}
+
+type RunnerOption func(config *RunnerConfig)
+
+func SkipTLS(skip *bool) RunnerOption {
+	return func(config *RunnerConfig) {
+		if skip != nil {
+			config.SkipTLS = *skip
+		}
+	}
+}
+
+func (r *RunnerConfig) apply(options []RunnerOption) {
+	for _, option := range options {
+		option(r)
+	}
+}
+
+func (r *ContainerCommandRunner) argsForCmd(cmd string, args... string) []string {
+	cmdArgs := []string{cmd}
+	switch r.containerTool {
+	case PodmanTool:
+		switch cmd {
+		case "build", "pull", "push", "login", "search":
+			// --tls-verify is a valid flag for these podman subcommands
+			if r.config.SkipTLS {
+				cmdArgs = append(cmdArgs, "--tls-verify=false")
+			}
+		}
+	case DockerTool:
+		if !r.config.SkipTLS {
+			cmdArgs = append(cmdArgs, "--tls")
+		}
+	default:
+	}
+	cmdArgs = append(cmdArgs, args...)
+	return cmdArgs
+}
+
+func defaultConfig(toolName string) *RunnerConfig {
+	switch toolName {
+	case "docker":
+		// docker disables tls verify by default, mimic that behavior
```

Comment:
Maybe I'm misunderstanding, but docker disables tls verify when the client is communicating with the daemon, ie when communicating over http to the daemon. It enables tls verify when pushing/pulling images, which I believe is the context here. Pushing to an insecure registry will by default cause docker to throw an error, without the flag set

Reply:
So, does setting `SkipTLSVerify=false` at the top-level in `opm` prevent the docker variant from communicating with the docker daemon?

Reply:
I was mistaken about this part, docker isn't blocked by the presence/absence of this option, it needs the --tls flag for client side auth, which we don't need to do.

Reply:
> Pushing to an insecure registry will by default cause docker to throw an error, without the flag set

Docker doesn't have any issues with an insecure local registry. Any remote registries should have tls on them, so it's a non-issue